test/onnx/dynamo/test_registry_dispatcher.py

```diff
@@ -0,0 +1,495 @@
+            self.dispatcher.get_aten_name(
+                self.unsupported_op_overloadpacket, self.diagnostic_context
+            )
+        with self.assertRaises(RuntimeError):
+            self.dispatcher.get_aten_name(
+                self.unsupported_op_node_target, self.diagnostic_context
+            )
+        with self.assertRaises(RuntimeError):
+            self.dispatcher.get_aten_name(
+                self.unsupported_builtin_op, self.diagnostic_context
+            )
+
+    def test_get_function_overloads(self):
+        # Test fall back to default op name
+        self.assertEqual(
+            self.dispatcher.get_function_overloads(
+                self.op_overload, "aten::add.Tensor", self.diagnostic_context
+            ),
+            self.dispatcher.get_function_overloads(
+                self.unsupported_op_overloadpacket, "aten::add", self.diagnostic_context
+            ),
+        )
+        with self.assertRaises(RuntimeError):
+            self.dispatcher.get_function_overloads(
+                self.unsupported_op_node_target, "aten::test", self.diagnostic_context
+            )
+
+    def test_warnings_in_find_the_perfect_or_nearest_match_onnxfunction(self):
+        with self.assertWarnsOnceRegex(
+            UserWarning,
+            "A perfect matched Opchema is not found in torchlib for aten::add",
+        ):
+            function_overloads = self.dispatcher.get_function_overloads(
+                self.op_overload, "aten::add", self.diagnostic_context
+            )
+            self.dispatcher._find_the_perfect_or_nearest_match_onnxfunction(
+                self.op_overload,
+                "aten::add",
+                function_overloads,
+                self.op_overload.args,
+                self.op_overload.kwargs,
+                self.diagnostic_context,
+            )
+
+    @common_utils.parametrize(
+        "available_opsets, target, expected",
+        [
+            ((7, 8, 9, 10, 11), 16, 11),
+            ((7, 8, 9, 10, 11), 11, 11),
+            ((7, 8, 9, 10, 11), 10, 10),
+            ((7, 8, 9, 10, 11), 9, 9),
+            ((7, 8, 9, 10, 11), 8, 8),
+            ((7, 8, 9, 10, 11), 7, 7),
+            ((9, 10, 16), 16, 16),
+            ((9, 10, 16), 15, 10),
+            ((9, 10, 16), 10, 10),
+            ((9, 10, 16), 9, 9),
+            ((9, 10, 16), 8, 9),
+            ((9, 10, 16), 7, 9),
+            ((7, 9, 10, 16), 16, 16),
+            ((7, 9, 10, 16), 10, 10),
+            ((7, 9, 10, 16), 9, 9),
+            ((7, 9, 10, 16), 8, 9),
+            ((7, 9, 10, 16), 7, 7),
+            ([17], 16, None),  # New op added in 17
+            ([9], 9, 9),
+            ([9], 8, 9),
+            ([], 16, None),
+            ([], 9, None),
+            ([], 8, None),
+            # Ops registered at opset 1 found as a fallback when target >= 9
+            ([1], 16, 1),
+        ],
+    )
+    def test_dispatch_opset_version_returns_correct_version(
```

Comment:
But do we actually need these tests knowing they will be removed?

Reply:
I wrote the tests couple weeks ago. I think as long as the new registry is not merged yet, we should keep this.